source/uwp/Renderer/lib/ActionHelpers.cpp

```diff
@@ -274,10 +276,25 @@ namespace AdaptiveNamespace::ActionHelpers
                         _In_ IAdaptiveRenderArgs* renderArgs,
                         _Outptr_ IUIElement** actionControl)
     {
-        // Render a button for the action
+        // determine what type of action we're building
         ComPtr<IAdaptiveActionElement> action(adaptiveActionElement);
-        ComPtr<IButton> button =
-            XamlHelpers::CreateXamlClass<IButton>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_Button));
+        ABI::AdaptiveNamespace::ActionType actionType;
+        RETURN_IF_FAILED(action->get_ActionType(&actionType));
+
+        // now construct an appropriate button for the action type
+        ComPtr<IButton> button;
+        if (actionType == ABI::AdaptiveNamespace::ActionType_OpenUrl)
+        {
+            // OpenUrl buttons should appear as links for accessibility purposes, so we use our custom LinkButton.
+            auto linkButton = winrt::make<LinkButton>();
+            button = linkButton.as<IButton>().detach();
```

Comment:
> button = linkButton.as<IButton>().detach [](start = 11, length = 41)

Can we just add some very basic unit tests to test construction etc of the new LinkButton object type? Also how much of manual testing should we try for this? ANy backcompat issues that you anticipate at all ?